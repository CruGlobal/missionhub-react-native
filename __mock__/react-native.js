import { NativeModules } from 'react-native';

// From https://github.com/facebook/react-native/issues/18279#issuecomment-374177940
// Will be made unnecessary once https://github.com/facebook/react-native/pull/18718 is released. It doesn't seem to be in 0.55.3
NativeModules.BlobModule = {
  ...NativeModules.BlobModule,
  addNetworkingHandler: jest.fn(),
};
