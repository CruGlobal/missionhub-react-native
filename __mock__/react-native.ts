jest.mock(
  'react-native/Libraries/Utilities/NativePlatformConstantsIOS',
  () => ({
    // @ts-ignore
    ...jest.requireActual(
      'react-native/Libraries/Utilities/NativePlatformConstantsIOS',
    ),
    getConstants: () => ({
      forceTouchAvailable: false,
      interfaceIdiom: 'en',
      isTesting: false,
      osVersion: 'ios',
      reactNativeVersion: { major: 60, minor: 1, patch: 0 },
      systemName: 'ios',
    }),
  }),
);
