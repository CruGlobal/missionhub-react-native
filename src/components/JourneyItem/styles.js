
import { StyleSheet } from 'react-native';
import theme from '../../theme';

export default StyleSheet.create({
  row: {
    paddingVertical: 19,
    paddingHorizontal: 24,
    backgroundColor: theme.white,
    width: theme.fullWidth,
  },
  icon: {
    color: theme.secondaryColor,
    paddingRight: 15,
  },
  textWrap: {
    borderLeftColor: theme.separatorColor,
    borderLeftWidth: 1,
    paddingLeft: 20,
    paddingBottom: 5,
  },
  date: {
    color: theme.inactiveColor,
    fontSize: 10,
  },
  text: {
    fontSize: 14,
  },
});
