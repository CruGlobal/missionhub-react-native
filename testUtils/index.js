import 'react-native';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Need to import for all translation screens to work properly
import '../src/i18n';

Enzyme.configure({ adapter: new Adapter() });

export const createMockStore = (state = {}) => {
  return {
    getState: jest.fn(() => (state)),
    dispatch: jest.fn((response) => Promise.resolve(response)),
    subscribe: jest.fn(),
  };
};

export const createMockNavState = (params = {}) => {
  return { state: { params } };
};

export const testSnapshot = (data) => {
  expect(renderer.create(data)).toMatchSnapshot();
};

export const renderShallow = (component, store) => {
  let renderedComponent = shallow(
    component,
    { context: { store: store } }
  );

  // If component has translation wrappers, dive deeper
  while (renderedComponent.is('Translate') || renderedComponent.is('I18n')) {
    renderedComponent = renderedComponent.dive();
  }

  // Render contents of component
  renderedComponent = renderedComponent.dive();
  return renderedComponent;
};

export const testSnapshotShallow = (component, store) => {
  const renderedComponent = renderShallow(component, store);
  expect(renderedComponent).toMatchSnapshot();
  return renderedComponent;
};

export const mockFnWithParams = (obj, method, expectedReturn, ...expectedParams) => {
  return obj[method] = jest.fn(
    (...actualParams) => {
      expect(actualParams).toEqual(expectedParams);
      return expectedReturn;
    }
  );
};
