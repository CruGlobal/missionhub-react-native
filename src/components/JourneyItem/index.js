import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';

import { Flex, Text, Icon, DateComponent } from '../common';
import styles from './styles';
import { INTERACTION_TYPES } from '../../constants';

const interactionsArr = Object.keys(INTERACTION_TYPES).map((key) => INTERACTION_TYPES[key]);

@translate('journeyItem')
export default class JourneyItem extends Component {
  setNativeProps(nProps) { this._view.setNativeProps(nProps); }
  renderDate() {
    const { item } = this.props;

    return <DateComponent date={item.date} style={styles.date} format="LL" />;
  }

  hasOldStage(item) {
    return item.old_pathway_stage.name;
  }

  translatableStage(item) {
    return { personName: item.personName, oldStage: item.old_pathway_stage.name, newStage: item.new_pathway_stage.name };
  }

  renderTitle() {
    const { t, item, type } = this.props;
    let title;
    if (type === 'step') {
      title = t('stepTitle');
    } else if (type === 'stage') {
      if (item.old_pathway_stage.name) {
        title = t('stageTitle', this.translatableStage(item));
      } else {
        title = item.new_pathway_stage.name;
      }
    } else if (type === 'survey' && item.survey) {
      title = item.survey.title;
    } else if (type === 'interaction') {
      const interaction = interactionsArr.find((i) => i.id === item.interaction_type_id);
      if (interaction) {
        title = t(interaction.translationKey);
      }
    }

    if (!title) return null;

    return (
      <Text style={styles.title}>
        {title}
      </Text>
    );
  }
  renderText() {
    const { t, item, type } = this.props;
    let text;
    if (type === 'step') {
      text = item.title;
      if (item.note) {
        text = `${text}\n\n${item.note}`;
      }
    } else if (type === 'stage') {
      if (this.hasOldStage(item)) {
        text = t('stageText', this.translatableStage(item));
      } else {
        text = t('stageStart', this.translatableStage(item));
      }
    } else {
      text = item.text;
    }

    return (
      <Text style={styles.text}>
        {text}
      </Text>
    );
  }

  renderIcon() {
    const { item, type } = this.props;
    let iconType;
    if (type === 'step') {
      iconType = 'stepsIcon';
    } else if (type === 'stage') {
      iconType = 'journeyIcon';
    } else if (type === 'survey') {
      iconType = 'surveyIcon';
    } else if (type === 'interaction') {
      const interaction = interactionsArr.find((i) => i.id === item.interaction_type_id);
      if (interaction) {
        iconType = interaction.iconName;
      }
    }

    if (!iconType) return null;

    return (
      <Icon
        name={iconType}
        type="MissionHub"
        size={32}
        style={styles.icon}
      />
    );
  }

  renderContent() {
    if (this.props.type === 'survey') return this.renderSurvey();
    return (
      <Flex value={1} direction="column" style={styles.textWrap}>
        {this.renderDate()}
        {this.renderTitle()}
        {this.renderText()}
      </Flex>
    );
  }

  renderSurvey() {
    const { answers } = this.props.item;
    return (
      <Flex value={1} direction="column" style={styles.textWrap}>
        {this.renderDate()}
        {this.renderTitle()}
        {
          answers.map((a) => (
            <Flex direction="column" key={a.id}>
              <Text style={styles.question}>{a.question.label}</Text>
              <Text style={styles.text}>{a.value}</Text>
            </Flex>
          ))
        }
      </Flex>
    );
  }

  render() {
    const { type } = this.props;
    return (
      <Flex
        ref={(c) => this._view = c}
        direction="row"
        align="center"
        style={[
          styles.row,
          type && styles[type] ? styles[type] : null,
        ]}
      >
        {this.renderIcon()}
        {this.renderContent()}
      </Flex>
    );
  }

}

JourneyItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    text: PropTypes.string,
    title: PropTypes.string,
    completed_at: PropTypes.date,
  }).isRequired,
  type: PropTypes.oneOf([ 'step', 'stage', 'survey', 'interaction' ]),
};
