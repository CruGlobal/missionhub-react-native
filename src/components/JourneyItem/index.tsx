import React, { Component } from 'react';
import PropTypes from 'prop-types';
import i18next from 'i18next';
import { withTranslation } from 'react-i18next';
import { Text } from 'react-native';

import { Flex, Icon, DateComponent } from '../common';
import { INTERACTION_TYPES, ACCEPTED_STEP } from '../../constants';
import { getAssignedByName, getAssignedToName } from '../../utils/common';
import { localizedStageSelector } from '../../selectors/stages';

import styles from './styles';

const interactionsArr = Object.keys(INTERACTION_TYPES).map(
  // @ts-ignore
  key => INTERACTION_TYPES[key],
);

// @ts-ignore
@withTranslation('journeyItem')
export default class JourneyItem extends Component {
  // @ts-ignore
  setNativeProps(nProps) {
    // @ts-ignore
    this._view.setNativeProps(nProps);
  }

  renderDate() {
    // @ts-ignore
    const { item } = this.props;
    return (
      <DateComponent
        date={
          item._type === ACCEPTED_STEP ? item.completed_at : item.created_at
        }
        style={styles.date}
        format="LL"
      />
    );
  }

  // @ts-ignore
  oldStage(item) {
    return localizedStageSelector(item.old_pathway_stage, i18next.language)
      .name;
  }

  // @ts-ignore
  translatableStage(item) {
    return {
      personName: item.person.first_name,
      oldStage: this.oldStage(item),
      newStage: localizedStageSelector(item.new_pathway_stage, i18next.language)
        .name,
    };
  }

  // @ts-ignore
  isSelfPathwayProgressionAudit(item) {
    // @ts-ignore
    return this.props.myId === item.person.id;
  }

  renderTitle() {
    const {
      // @ts-ignore
      t,
      // @ts-ignore
      item,
      // @ts-ignore
      myId,
      // @ts-ignore
      personFirstName,
      // @ts-ignore
      item: { _type },
    } = this.props;
    let title;
    if (_type === ACCEPTED_STEP) {
      const pathwayStage =
        (item.challenge_suggestion &&
          item.challenge_suggestion.pathway_stage &&
          ` ${
            localizedStageSelector(
              item.challenge_suggestion.pathway_stage,
              i18next.language,
            ).name
          } `) ||
        ' ';
      title = t('stepTitle', { stageName: pathwayStage });
    } else if (_type === 'pathway_progression_audit') {
      if (this.oldStage(item)) {
        title = t('stageTitle', this.translatableStage(item));
      } else {
        title = localizedStageSelector(item.new_pathway_stage, i18next.language)
          .name;
      }
    } else if (_type === 'answer_sheet' && item.survey) {
      title = item.survey.title;
    } else if (_type === 'interaction') {
      const interaction = interactionsArr.find(
        i => i.id === `${item.interaction_type_id}`,
      );
      if (interaction) {
        title = t(interaction.translationKey);
      }
    } else if (_type === 'contact_assignment') {
      const assignedToName = getAssignedToName(myId, item);
      const assignedByName = getAssignedByName(myId, item);

      title = t('contactAssignment', {
        assignedByName,
        assignedContactName: personFirstName,
        assignedToName,
      });
    } else if (_type === 'contact_unassignment') {
      const assignedToName = getAssignedToName(myId, item);

      title = t('contactUnassignment', {
        assignedContactName: personFirstName,
        assignedToName,
      });
    }

    if (!title) {
      return null;
    }

    return <Text style={styles.title}>{title}</Text>;
  }

  renderText() {
    const {
      // @ts-ignore
      t,
      // @ts-ignore
      item,
      // @ts-ignore
      item: { _type },
    } = this.props;
    let text;
    if (_type === ACCEPTED_STEP) {
      text = item.title;
      if (item.note) {
        text = `${text}\n\n${item.note}`;
      }
    } else if (_type === 'pathway_progression_audit') {
      if (this.oldStage(item)) {
        if (this.isSelfPathwayProgressionAudit(item)) {
          text = t('stageTextSelf', this.translatableStage(item));
        } else {
          text = t('stageText', this.translatableStage(item));
        }
      } else {
        if (this.isSelfPathwayProgressionAudit(item)) {
          text = t('stageStartSelf', this.translatableStage(item));
        } else {
          text = t('stageStart', this.translatableStage(item));
        }
      }
    } else if (_type === 'interaction') {
      text = item.comment;
    }

    if (!text) {
      return null;
    }

    return <Text style={styles.text}>{text}</Text>;
  }

  renderIcon() {
    const {
      // @ts-ignore
      item,
      // @ts-ignore
      item: { _type },
    } = this.props;
    let iconType;
    if (_type === ACCEPTED_STEP) {
      iconType = 'stepsIcon';
    } else if (_type === 'pathway_progression_audit') {
      iconType = 'journeyIcon';
    } else if (_type === 'answer_sheet') {
      iconType = 'surveyIcon';
    } else if (_type === 'interaction') {
      const interaction = interactionsArr.find(
        i => i.id === `${item.interaction_type_id}`,
      );
      if (interaction) {
        iconType = interaction.iconName;
      }
    } else if (
      _type === 'contact_assignment' ||
      _type === 'contact_unassignment'
    ) {
      iconType = 'statusIcon';
    }

    if (!iconType) {
      return null;
    }

    return (
      <Flex value={1}>
        <Icon
          name={iconType}
          type="MissionHub"
          size={32}
          style={[
            styles.icon,
            iconType === 'commentIcon' ? styles.commentIcon : {},
          ]}
        />
      </Flex>
    );
  }

  renderContent() {
    // @ts-ignore
    if (this.props.item._type === 'answer_sheet') {
      return this.renderSurvey();
    }
    return (
      <Flex value={3.5} direction="column" style={styles.textWrap}>
        {this.renderDate()}
        {this.renderTitle()}
        {this.renderText()}
      </Flex>
    );
  }

  renderSurvey() {
    // @ts-ignore
    const { answers } = this.props.item;
    return (
      <Flex value={3.5} direction="column" style={styles.textWrap}>
        {this.renderDate()}
        {this.renderTitle()}
        {/* 
        // @ts-ignore */}
        {answers.map(a => (
          <Flex direction="column" key={a.id}>
            <Text style={styles.question}>{a.question.label}</Text>
            <Text style={styles.text}>{a.value}</Text>
          </Flex>
        ))}
      </Flex>
    );
  }

  // @ts-ignore
  ref = c => (this._view = c);

  render() {
    return (
      <Flex ref={this.ref} direction="row" align="center" style={styles.row}>
        {this.renderIcon()}
        {this.renderContent()}
      </Flex>
    );
  }
}

// @ts-ignore
JourneyItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    _type: PropTypes.oneOf([
      ACCEPTED_STEP,
      'pathway_progression_audit',
      'answer_sheet',
      'interaction',
      'contact_assignment',
      'contact_unassignment',
    ]),
    text: PropTypes.string,
    title: PropTypes.string,
    // @ts-ignore
    completed_at: PropTypes.date,
  }).isRequired,
  myId: PropTypes.string.isRequired,
  personFirstName: PropTypes.string.isRequired,
};
