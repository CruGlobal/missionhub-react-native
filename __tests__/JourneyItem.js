import 'react-native';
import React from 'react';

import JourneyItem from '../src/components/JourneyItem';
import { testSnapshotShallow } from '../testUtils';

const myId = '484893';
const person = {
  id: '889433',
  _type: 'person',
  first_name: 'Test Person',
};
const date = '2017-12-06T14:24:52Z';

describe('step', () => {
  const mockStep = {
    id: '1',
    _type: 'accepted_challenge',
    title: 'Test Step',
    completed_at: date,
  };
  const challenge_suggestion = {
    pathway_stage: {
      name: 'Guiding',
    },
  };

  it('is rendered correctly without comment', () => {
    testSnapshotShallow(
      <JourneyItem
        item={mockStep}
        myId={myId} />
    );
  });

  it('is rendered correctly with comment', () => {
    testSnapshotShallow(
      <JourneyItem
        item={{ ...mockStep, note: 'test comment on completed step' }}
        myId={myId} />
    );
  });

  it('is rendered correctly with pathway stage', () => {
    testSnapshotShallow(
      <JourneyItem
        item={{ ...mockStep, challenge_suggestion }}
        myId={myId} />
    );
  });
});

describe('stage', () => {
  const mockStageProgression = {
    id: '3',
    _type: 'pathway_progression_audit',
    comment: 'Test Stage Change',
    old_pathway_stage: { id: '1', _type: 'pathway_stage', name: 'Uninterested' },
    new_pathway_stage: { id: '2', _type: 'pathway_stage', name: 'Curious' },
    created_at: date,
    person,
  };

  it('is rendered correctly with old stage for a contact', () => {
    testSnapshotShallow(
      <JourneyItem
        item={mockStageProgression}
        myId={myId} />
    );
  });

  it('is rendered correctly with old stage for self', () => {
    testSnapshotShallow(
      <JourneyItem
        item={mockStageProgression}
        myId={person.id} />
    );
  });

  it('is rendered correctly without old stage for a contact', () => {
    testSnapshotShallow(
      <JourneyItem
        item={{
          ...mockStageProgression,
          old_pathway_stage: {},
        }}
        myId={myId} />

    );
  });

  it('is rendered correctly without old stage for self', () => {
    testSnapshotShallow(
      <JourneyItem
        item={{
          ...mockStageProgression,
          old_pathway_stage: {},
        }}
        myId={person.id} />

    );
  });
});

it('renders survey correctly', () => {
  testSnapshotShallow(
    <JourneyItem
      item={{
        id: '5',
        _type: 'answer_sheet',
        survey: { title: 'Test Survey' },
        answers: [
          { id: '1', question: { label: 'Question 1' }, value: 'Answer 1' },
          { id: '2', question: { label: 'Question 2' }, value: 'Answer 2' },
          { id: '3', question: { label: 'Question 3' }, value: 'Answer 3' },
        ],
        created_at: date,
      }}
      myId={myId} />
  );
});

it('renders interaction correctly', () => {
  testSnapshotShallow(
    <JourneyItem
      item={{
        id: '4',
        _type: 'interaction',
        interaction_type_id: 1,
        comment: 'Test Interaction',
        initiators: [
          { id: myId },
        ],
        organization: null,
        created_at: date,
      }}
      myId={myId} />
  );
});
