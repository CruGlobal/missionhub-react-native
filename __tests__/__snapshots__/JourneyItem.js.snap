// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders interaction correctly 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "paddingRight": 24,
          "paddingVertical": 19,
          "width": 750,
        },
        null,
      ],
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#ffffff",
            },
            Object {
              "fontSize": 32,
            },
            Array [
              Object {
                "alignSelf": "center",
                "color": "#007398",
                "fontSize": 40,
                "textAlign": "center",
              },
              Object {
                "fontSize": 32,
              },
            ],
          ],
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#E6E8EC",
          "borderLeftWidth": 1,
          "paddingBottom": 5,
          "paddingLeft": 20,
        },
        Object {
          "flex": 3.5,
          "flexDirection": "column",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "color": "#A0A2A6",
            "fontSize": 10,
          },
        ]
      }
    >
      December 6, 2017
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "color": "#007398",
            "fontSize": 16,
          },
        ]
      }
    >
      Comment
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    >
      Test Journey
    </Text>
  </View>
</View>
`;

exports[`renders survey correctly 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "paddingRight": 24,
          "paddingVertical": 19,
          "width": 750,
        },
        null,
      ],
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#ffffff",
            },
            Object {
              "fontSize": 32,
            },
            Array [
              Object {
                "alignSelf": "center",
                "color": "#007398",
                "fontSize": 40,
                "textAlign": "center",
              },
              Object {},
            ],
          ],
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#E6E8EC",
          "borderLeftWidth": 1,
          "paddingBottom": 5,
          "paddingLeft": 20,
        },
        Object {
          "flex": 3.5,
          "flexDirection": "column",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "color": "#A0A2A6",
            "fontSize": 10,
          },
        ]
      }
    >
      December 6, 2017
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "color": "#007398",
            "fontSize": 16,
          },
        ]
      }
    >
      Survey Test
    </Text>
    <View
      style={
        Array [
          Object {},
          Object {
            "flexDirection": "column",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "color": "#A0A2A6",
              "fontSize": 14,
              "marginTop": 8,
            },
          ]
        }
      >
        Question 1
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        Answer 1
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "flexDirection": "column",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "color": "#A0A2A6",
              "fontSize": 14,
              "marginTop": 8,
            },
          ]
        }
      >
        Question 2
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        Answer 2
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "flexDirection": "column",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "color": "#A0A2A6",
              "fontSize": 14,
              "marginTop": 8,
            },
          ]
        }
      >
        Question 3
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        Answer 3
      </Text>
    </View>
  </View>
</View>
`;

exports[`stage is rendered correctly with old stage 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "paddingRight": 24,
          "paddingVertical": 19,
          "width": 750,
        },
        null,
      ],
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#ffffff",
            },
            Object {
              "fontSize": 32,
            },
            Array [
              Object {
                "alignSelf": "center",
                "color": "#007398",
                "fontSize": 40,
                "textAlign": "center",
              },
              Object {},
            ],
          ],
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#E6E8EC",
          "borderLeftWidth": 1,
          "paddingBottom": 5,
          "paddingLeft": 20,
        },
        Object {
          "flex": 3.5,
          "flexDirection": "column",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "color": "#A0A2A6",
            "fontSize": 10,
          },
        ]
      }
    >
      December 6, 2017
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "color": "#007398",
            "fontSize": 16,
          },
        ]
      }
    >
      Uninterested to Curious
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    >
      Test Person changed from Uninterested to Curious
    </Text>
  </View>
</View>
`;

exports[`stage is rendered correctly without old stage 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "paddingRight": 24,
          "paddingVertical": 19,
          "width": 750,
        },
        null,
      ],
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#ffffff",
            },
            Object {
              "fontSize": 32,
            },
            Array [
              Object {
                "alignSelf": "center",
                "color": "#007398",
                "fontSize": 40,
                "textAlign": "center",
              },
              Object {},
            ],
          ],
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#E6E8EC",
          "borderLeftWidth": 1,
          "paddingBottom": 5,
          "paddingLeft": 20,
        },
        Object {
          "flex": 3.5,
          "flexDirection": "column",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "color": "#A0A2A6",
            "fontSize": 10,
          },
        ]
      }
    >
      December 6, 2017
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "color": "#007398",
            "fontSize": 16,
          },
        ]
      }
    >
      Curious
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    >
      You added Test Person to MissionHub as Curious
    </Text>
  </View>
</View>
`;

exports[`step is rendered correctly with comment 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "paddingRight": 24,
          "paddingVertical": 19,
          "width": 750,
        },
        null,
      ],
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#ffffff",
            },
            Object {
              "fontSize": 32,
            },
            Array [
              Object {
                "alignSelf": "center",
                "color": "#007398",
                "fontSize": 40,
                "textAlign": "center",
              },
              Object {},
            ],
          ],
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#E6E8EC",
          "borderLeftWidth": 1,
          "paddingBottom": 5,
          "paddingLeft": 20,
        },
        Object {
          "flex": 3.5,
          "flexDirection": "column",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "color": "#A0A2A6",
            "fontSize": 10,
          },
        ]
      }
    >
      December 6, 2017
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "color": "#007398",
            "fontSize": 16,
          },
        ]
      }
    >
      Growing Step of Faith
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    >
      undefined

test comment on completed step
    </Text>
  </View>
</View>
`;

exports[`step is rendered correctly without comment 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "paddingRight": 24,
          "paddingVertical": 19,
          "width": 750,
        },
        null,
      ],
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#ffffff",
            },
            Object {
              "fontSize": 32,
            },
            Array [
              Object {
                "alignSelf": "center",
                "color": "#007398",
                "fontSize": 40,
                "textAlign": "center",
              },
              Object {},
            ],
          ],
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#E6E8EC",
          "borderLeftWidth": 1,
          "paddingBottom": 5,
          "paddingLeft": 20,
        },
        Object {
          "flex": 3.5,
          "flexDirection": "column",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "color": "#A0A2A6",
            "fontSize": 10,
          },
        ]
      }
    >
      December 6, 2017
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "color": "#007398",
            "fontSize": 16,
          },
        ]
      }
    >
      Growing Step of Faith
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#505256",
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    />
  </View>
</View>
`;
